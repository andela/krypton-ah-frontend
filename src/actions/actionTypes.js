const actionTypes = {
  SIGNUP: 'SIGNUP',
  UPDATE_FIELD: 'UPDATE_FIELD'
};

export default actionTypes;
