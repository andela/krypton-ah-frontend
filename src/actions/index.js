export const START_FETCH_ARTICLES = 'START_FETCHING_ARTICLES';
export const FETCH_ARTICLES = 'FETCH_ARTICLES';
