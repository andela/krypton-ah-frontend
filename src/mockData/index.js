import { API_BASE_URL } from '../constants';

const items = [
  {
    description:
      'The Touch Bar replaces the function keys that have long occupied the top of your keyboard with something much more versatile and capable.16 It changes automatically based on what youâ€™re doing to show you relevant The Touch Bar replaces the function keys that have long occupied the top of your keyboard with something much more versatile and capable.16 It changes automatically based',
    shortdescription:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach, while an optimist drains the barrel, fishes out all the apples and makes pie',
    shorterdescription:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  }
];

const categories = ['Politics', 'Family', 'Techonology', 'Health', 'Sports', 'Wellbeing'];

const newCategories = [
  { key: 1, name: 'Politics' },
  { key: 2, name: 'Family' },
  { key: 3, name: 'Techonology' }
];

const popularArticles = [
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist settles for the first one within reach',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  }
];

const mockHeader = 'this is a placeholder header';
const trendingArticles = [
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Trending Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Trending Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Trending Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  },
  {
    description:
      'When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ',
    content: "Apple's news at CES 2019 shows it's facing some hard truths",
    title: 'Popular Title'
  }
];

const payload = {
  firstname: 'john',
  lastname: 'joseph',
  email: 'jo@jos.com',
  password: 'password123R'
};
const fakeUser2 = {
  email: 'jo@jos.com',
  password: 'password'
};
const signupEndpoint = `${API_BASE_URL}/users/signup`;
const signinEndpoint = `${API_BASE_URL}/users/signin`;

const signupOkResponse = {
  status: 201,
  success: true,
  data: {
    message:
      'Account successfully created, Kindly check your email to activate your account. In case you did not receive the activation link in your mail, kindly visit this link localhost:3000/api/v1/users/resend/activation/mail to resend the mail.'
  }
};

const mockResponse = {
  message: 'Successfully logged in'
};
const error = {
  response: {
    status: 400,
    success: false,
    data: {
      message: 'Oops! something went wrong, kindly try again'
    }
  }
};
const loginOkResponse = {
  status: 200,
  success: true,
  data: {
    token: 'sampleToken',
    message: 'You have sccessfully login'
  }
};
const loginBadResponse = {
  status: 400,
  success: false,
  data: {
    message: 'Incorrect credentials'
  }
};

const startAction = [{ type: 'START' }];

const signupSuccessAction = [
  {
    type: 'SIGNUP_SUCCESS',
    payload
  }
];
const signupFailureAction = [
  {
    type: 'SIGNUP_FAILURE',
    payload
  }
];

const fetchArticlesresponse = {
  data: {
    data:
    [{}, {}, {}, {}, {}, {}, {}, {}, {}]
  }
};

const fetchArticles = {
  data: [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}]
};
const socialToken = 'hyakfhkdhiuashvk98924cxndusjdusnfixomj9';

const mockPath = 'auth/facebook/';
const profileData = {
  firstname: 'Isaiah',
  lastname: 'Afolayan',
  profileImage: '',
  bio: 'Software developer at Andela',
  id: 'd1564aa6-e81d-41ab-8469-0aa573f4a6c5',
  country: 'Nigeria',
  phonenumber: '+2348145692034',
  gender: 'male',
  username: 'Isaiah'
};

const autoLoginResponse = {
  data: {
    message: 'Your account has been successfully activated! You can begin spreaing your ideas now.'
  }
};
const getUserFailureResponse = {
  data: {
    success: false,
    message: 'error retieving user'
  }
};

const getUserSuccessResponse = {
  data: {
    success: true,
    message: 'users retrieved succesfully',
    data: {
      id: 'd1564aa6-e81d-41ab-8469-0aa573f4a6c5',
      email: '1skyboat1@gmail.com',
      password: '$2b$08$Q5LSk9wYgQeuxtSTGeq/0eBMclMVg1B7wE/hfWRCjsUb9YHkgx4h.',
      firstname: 'firstname',
      lastname: 'lastname',
      isverified: true,
      createdAt: '2019-02-08T09:58:13.163Z',
      updatedAt: '2019-02-08T09:58:13.163Z',
      userprofile: {
        id: 1,
        avatar: null,
        bio: 'A world-class software developer',
        UserId: 'd1564aa6-e81d-41ab-8469-0aa573f4a6c5',
        username: 'eltneg',
        phonenumber: '+2348137457862',
        country: 'Nigeria',
        gender: 'male',
        emailnotification: false,
        createdAt: '2019-02-14T16:54:58.292Z',
        updatedAt: '2019-02-18T08:41:31.199Z'
      },
      articles: [
        {
          title: 'Article title6',
          description: 'valid article description',
          isPublished: false,
          createdAt: '2019-02-14T17:53:01.904Z',
          id: 'a76d2040-7510-43d8-b58b-a32aba67ab48'
        }
      ]
    }
  }
};

const jwtToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlMb2FkIjp7ImlkIjoiZDE1NjRhYTYtZTgxZC00MWFiLTg0NjktMGFhNTczZjRhNmM1Iiwicm9sZSI6W119LCJpYXQiOjE1NTEzNDg0NjEsImV4cCI6MTU1MTQzNDg2MX0.W-Q3vDCP1b4iKMOm_cBSMTzmRqvU9kpzcP-E2ka2n4U';

export {
  fetchArticles,
  payload,
  fakeUser2,
  signupEndpoint,
  signinEndpoint,
  mockResponse,
  signupOkResponse,
  startAction,
  signupSuccessAction,
  signupFailureAction,
  error,
  loginOkResponse,
  loginBadResponse,
  items,
  categories,
  newCategories,
  popularArticles,
  trendingArticles,
  mockHeader,
  socialToken,
  mockPath,
  profileData,
  jwtToken,
  autoLoginResponse,
  fetchArticlesresponse,
  getUserSuccessResponse,
  getUserFailureResponse
};
