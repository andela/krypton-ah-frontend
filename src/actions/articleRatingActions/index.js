import { toast } from 'react-toastify';
import actionTypes from './actionTypes';
import { getUserRating, createRating } from '../../helpers/axiosHelper/articleRating';
import triggerLoading from '../authAction/loading';

const {
  CREATE_ARTICLE_RATING_LOADING,
  CREATE_ARTICLE_RATING_SUCCESS,
  CREATE_ARTICLE_RATING_FAILURE,
  GET_USER_RATING_LOADING,
  GET_USER_RATING_SUCCESS,
  GET_USER_RATING_FAILURE
} = actionTypes;

const createRatingSuccess = payload => ({
  type: CREATE_ARTICLE_RATING_SUCCESS,
  payload: payload.data
});

const createRatingFailure = payload => ({
  type: CREATE_ARTICLE_RATING_FAILURE,
  payload: payload.data
});

const getUserRatingSuccess = payload => ({
  type: GET_USER_RATING_SUCCESS,
  payload: payload.data
});

const getUserRatingFailure = payload => ({
  type: GET_USER_RATING_FAILURE,
  payload: payload.data
});

const createArticleRating = ratingdetails => async (dispatch) => {
  try {
    dispatch(triggerLoading(CREATE_ARTICLE_RATING_LOADING));
    const response = await createRating(ratingdetails);
    dispatch(createRatingSuccess(response));
    toast.success(response.data.message);
  } catch (error) {
    dispatch(createRatingFailure(error.response));
  }
};

const getUserArticleRating = articleId => async (dispatch) => {
  try {
    dispatch(triggerLoading(GET_USER_RATING_LOADING));
    const response = await getUserRating(articleId);
    dispatch(getUserRatingSuccess(response));
  } catch (error) {
    dispatch(getUserRatingFailure(error.response));
  }
};

export {
  createArticleRating,
  getUserArticleRating,
  createRatingSuccess,
  createRatingFailure,
  getUserRatingSuccess,
  getUserRatingFailure
};
