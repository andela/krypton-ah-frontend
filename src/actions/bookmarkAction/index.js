import { toast } from 'react-toastify';
import { createBookmark, removeBookmark } from '../../helpers/axiosHelper/bookmark';
import actionTypes from '../readArticleAction/actionTypes';

const { TOGGLE_ARTICLE_BOOKMARK } = actionTypes;

export const bookmarkSuccess = () => ({
  type: TOGGLE_ARTICLE_BOOKMARK
});

export const bookmarkedAction = bookmarkDetail => async (dispatch) => {
  try {
    const response = await createBookmark(bookmarkDetail);
    dispatch(bookmarkSuccess());
    toast.success(response.data.message);
  } catch (error) {
    toast.error(error.response.data.message);
  }
};

export const removeBookmarkSuccess = () => ({
  type: TOGGLE_ARTICLE_BOOKMARK
});

export const removeBookmarkAction = bookmarkDetail => async (dispatch) => {
  try {
    const response = await removeBookmark(bookmarkDetail);
    dispatch(removeBookmarkSuccess());
    toast.success(response.data.message);
  } catch (error) {
    toast.error(error.response.data.message);
  }
};
