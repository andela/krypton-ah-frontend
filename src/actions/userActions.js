import actionTypes from './actionTypes';

const { SIGNUP, UPDATE_FIELD } = actionTypes;

export const doSignUp = payload => (dispatch) => {
  dispatch({
    type: SIGNUP,
    payload
  });
};

export const doUpdateUserField = payload => (dispatch) => {
  dispatch({
    type: UPDATE_FIELD,
    payload
  });
};
