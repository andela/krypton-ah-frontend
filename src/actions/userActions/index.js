import { getUser, updateProfile, createProfile } from '../../helpers/axiosHelper/user';
import triggerLoading from '../authAction/loading';
import actionTypes from './actionTypes';

const {
  FETCH_USER_LOADING,
  UPDATE_USER_LOADING,
  FETCH_USER_SUCCESS,
  FETCH_USER_FAILURE,
  UPDATE_USER_SUCCESS,
  UPDATE_USER_FAILURE
} = actionTypes;

export const fetchUserSuccess = userData => ({
  type: FETCH_USER_SUCCESS,
  payload: userData
});

export const fetchUserFailure = () => ({
  type: FETCH_USER_FAILURE
});

export const updateUserSuccess = userData => ({
  type: UPDATE_USER_SUCCESS,
  payload: userData
});

export const updateUserFailure = () => ({
  type: UPDATE_USER_FAILURE
});

export const fetchUser = userId => async (dispatch) => {
  try {
    dispatch(triggerLoading(FETCH_USER_LOADING));
    const response = await getUser(userId);
    if (response.data.success) {
      dispatch(fetchUserSuccess(response.data.data));
    } else {
      dispatch(fetchUserFailure());
    }
  } catch (error) {
    dispatch(fetchUserFailure());
  }
};

export const updateUserProfile = (payload, create) => async (dispatch) => {
  try {
    dispatch(triggerLoading(UPDATE_USER_LOADING));
    const response = create ? await createProfile(payload) : await updateProfile(payload);
    if (response.data.success) {
      dispatch(updateUserSuccess(response.data.data));
    } else {
      dispatch(updateUserFailure());
    }
  } catch (error) {
    dispatch(updateUserFailure());
  }
};
