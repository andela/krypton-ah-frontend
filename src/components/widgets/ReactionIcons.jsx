import React, { Component, Fragment } from 'react';
import { Icon } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { dateFormatter } from '../../helpers/articleInfoFormatter';
import { getTotalReactions, articleReaction } from '../../actions/articleReactionsAction';

class ReactionIcons extends Component {
  state = {
    likeClicked: false,
    dislikeClicked: false
  };

  componentDidMount = () => {
    this.props.getTotalReactions(this.props.selectedArticleId);
  };

  createLikeIcon = (outline = 'outline') => {
    const name = outline.length === 0 ? 'thumbs up' : `thumbs up ${outline}`;
    return (
      <i role="presentation" onClick={this.createReaction.bind(this, 'like')}>
        <Icon disabled link size="small" fitted name={`thumbs up ${name}`} />
      </i>
    );
  };

  createDislikeIcon = (outline = 'outline') => {
    const name = outline.length === 0 ? 'thumbs down' : `thumbs down ${outline}`;
    return (
      <i role="presentation" onClick={this.createReaction.bind(this, 'dislike')}>
        <Icon disabled link size="small" fitted name={`thumbs down ${name}`} />
      </i>
    );
  };

  createReaction = (reaction) => {
    this.props.articleReaction(this.props.selectedArticleId, reaction);
  };

  render() {
    const { date, numberofcomments, totalArticleReactions } = this.props;
    const { likes, dislikes } = totalArticleReactions.successResponse;
    const { likeClicked, dislikeClicked } = this.state;
    const outline = '';
    return (
      <Fragment>
        <Icon className="articleDate" disabled link size="small" fitted name="time">
          <span>{dateFormatter(date)}</span>
        </Icon>
        {likeClicked ? this.createLikeIcon(outline) : this.createLikeIcon()}
        {likes}
        {dislikeClicked ? this.createDislikeIcon(outline) : this.createDislikeIcon()}
        {dislikes}
        <i className="articleComment">
          <Icon disabled link size="small" fitted name="comments outline" />
          <span>{numberofcomments}</span>
        </i>
      </Fragment>
    );
  }
}
ReactionIcons.defaultProps = {
  date: '28th february 2019',
  numberofcomments: 23
};

ReactionIcons.propTypes = {
  date: PropTypes.string,
  numberofcomments: PropTypes.number,
  selectedArticleId: PropTypes.string.isRequired,
  getTotalReactions: PropTypes.func.isRequired,
  totalArticleReactions: PropTypes.object.isRequired,
  articleReaction: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  totalArticleReactions: state.totalArticleReactions
});

export { ReactionIcons as ReactionIconsPage };
export default connect(
  mapStateToProps,
  { getTotalReactions, articleReaction }
)(ReactionIcons);
