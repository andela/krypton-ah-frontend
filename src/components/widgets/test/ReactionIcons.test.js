import React from 'react';
import { shallow } from 'enzyme';
import { ReactionIconsPage } from '../ReactionIcons';
import { props } from '../../../mockData/readArticle';
import { getTotalReactions } from '../../../actions/articleReactionsAction';

describe('Test for reaction Icons', () => {
  it('should have initial state of false and change when clicked ', () => {
    const articleReaction = jest.fn();
    const wrapper = shallow(
      <ReactionIconsPage
        getTotalReactions={getTotalReactions}
        selectedArticleId={props.match.params.title}
        totalArticleReactions={props.totalArticleReactions}
        articleReaction={articleReaction}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should have initial state of false and change when clicked ', () => {
    const articleReaction = jest.fn();
    const wrapper = shallow(
      <ReactionIconsPage
        getTotalReactions={getTotalReactions}
        selectedArticleId={props.match.params.title}
        totalArticleReactions={props.totalArticleReactions}
        articleReaction={articleReaction}
      />
    );

    expect(wrapper).toMatchSnapshot();
    wrapper.instance().createReaction();
  });
});
