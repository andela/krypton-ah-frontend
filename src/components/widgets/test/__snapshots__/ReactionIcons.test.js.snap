// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for reaction Icons should have initial state of false and change when clicked  1`] = `
<Fragment>
  <Icon
    as="i"
    className="articleDate"
    disabled={true}
    fitted={true}
    link={true}
    name="time"
    size="small"
  >
    <span>
      Invalid Date
    </span>
  </Icon>
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs up thumbs up outline"
      size="small"
    />
  </i>
  2
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs down thumbs down outline"
      size="small"
    />
  </i>
  1
  <i
    className="articleComment"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="comments outline"
      size="small"
    />
    <span>
      23
    </span>
  </i>
</Fragment>
`;

exports[`Test for reaction Icons should have initial state of false and change when clicked  2`] = `
<Fragment>
  <Icon
    as="i"
    className="articleDate"
    disabled={true}
    fitted={true}
    link={true}
    name="time"
    size="small"
  >
    <span>
      Invalid Date
    </span>
  </Icon>
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs up thumbs up outline"
      size="small"
    />
  </i>
  2
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs down thumbs down outline"
      size="small"
    />
  </i>
  1
  <i
    className="articleComment"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="comments outline"
      size="small"
    />
    <span>
      23
    </span>
  </i>
</Fragment>
`;
